drivers/usb/udc/udc_smartbond.c

```diff
@@ -0,0 +1,1764 @@
+			for (int ep_idx = 1; ep_idx < EP_MAX; ++ep_idx) {
+				ep_state = usb_dc_get_ep_out_state(dev_state, ep_idx);
+				if (!ep_state->config.stat.halted) {
+					start_rx_packet(dev_state, ep_state);
+				}
+			}
+			udc_submit_event(dev_state->dev, UDC_EVT_RESUME, 0);
+		}
+	} else if (GET_BIT(alt_ev, USB_USB_ALTEV_REG_USB_SD3)) {
+		set_nfsr(dev_state, NFSR_NODE_SUSPEND);
+		USB->USB_ALTMSK_REG =
+			USB_USB_ALTMSK_REG_USB_M_RESET_Msk | USB_USB_ALTMSK_REG_USB_M_RESUME_Msk;
+		usb_clock_off(dev_state);
+		udc_submit_event(dev_state->dev, UDC_EVT_SUSPEND, 0);
+	}
+}
+
+static void udc_smartbond_isr(const struct device *dev)
+{
+	struct usb_dc_state *dev_state = dev->data;
+	uint32_t int_status = USB->USB_MAEV_REG & USB->USB_MAMSK_REG;
+
+	if (GET_BIT(int_status, USB_USB_MAEV_REG_USB_WARN)) {
+		handle_fifo_warning(dev_state);
+	}
+
+	if (GET_BIT(int_status, USB_USB_MAEV_REG_USB_CH_EV)) {
+		/* For now just clear interrupt */
+		(void)USB->USB_CHARGER_STAT_REG;
+	}
+
+	if (GET_BIT(int_status, USB_USB_MAEV_REG_USB_EP0_TX)) {
+		handle_ep0_tx(dev_state);
+	}
+
+	if (GET_BIT(int_status, USB_USB_MAEV_REG_USB_EP0_RX)) {
+		handle_ep0_rx(dev_state);
+	}
+
+	if (GET_BIT(int_status, USB_USB_MAEV_REG_USB_EP0_NAK)) {
+		handle_ep0_nak(dev_state);
+	}
+
+	if (GET_BIT(int_status, USB_USB_MAEV_REG_USB_RX_EV)) {
+		handle_rx_ev(dev_state);
+	}
+
+	if (GET_BIT(int_status, USB_USB_MAEV_REG_USB_NAK)) {
+		(void)USB->USB_NAKEV_REG;
+	}
+
+	if (GET_BIT(int_status, USB_USB_MAEV_REG_USB_FRAME)) {
+		if (dev_state->nfsr == NFSR_NODE_RESET) {
+			/*
+			 * During reset FRAME interrupt is enabled to periodically
+			 * check when reset state ends.
+			 * FRAME interrupt is generated every 1ms without host sending
+			 * actual SOF.
+			 */
+			check_reset_end(dev_state, USB_USB_ALTEV_REG_USB_RESET_Msk);
+		} else if (dev_state->nfsr == NFSR_NODE_WAKING) {
+			/* No need to call set_nfsr, just set state */
+			dev_state->nfsr = NFSR_NODE_WAKING2;
+		} else if (dev_state->nfsr == NFSR_NODE_WAKING2) {
+			/* No need to call set_nfsr, just set state */
+			dev_state->nfsr = NFSR_NODE_RESUME;
+			LOG_DBG("dev_state->nfsr = NFSR_NODE_RESUME %02x", USB->USB_MAMSK_REG);
+		} else if (dev_state->nfsr == NFSR_NODE_RESUME) {
+			set_nfsr(dev_state, NFSR_NODE_OPERATIONAL);
+			if (dev_state->ep_state[0][0].buf != NULL) {
+				USB->USB_MAMSK_REG |= USB_USB_MAMSK_REG_USB_M_EP0_RX_Msk;
+			}
+			LOG_DBG("Set operational %02x", USB->USB_MAMSK_REG);
+		} else {
+			USB->USB_MAMSK_REG &= ~USB_USB_MAMSK_REG_USB_M_FRAME_Msk;
+		}
+	}
+
+	if (GET_BIT(int_status, USB_USB_MAEV_REG_USB_TX_EV)) {
+		handle_tx_ev(dev_state);
+	}
+
+	if (GET_BIT(int_status, USB_USB_MAEV_REG_USB_ALT)) {
+		handle_alt_ev(dev_state);
+	}
+}
+
+/**
+ * USB functionality can be disabled from HOST and DEVICE side.
+ * Host side is indicated by VBUS line.
+ * Device side is decided by pair of calls usb_dc_attach()/usb_dc_detach,
+ * USB will only work when application calls usb_dc_attach() and VBUS is present.
```

Comment:
usb_dc_attach()?

Reply:
change to reflect API calls udc_enable/udc_disable